### PR TITLE
Feature collection response

### PR DESCRIPTION
**Pull request by jrambla**
Schemas & examples for `beaconCollectionResponse `added

### CODE REVIEW

File: responses/examples-fullDocuments/beaconCollectionsResponse-MID-example.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+  "$schema": "../beaconCollectionsResponse.json",
+  "meta": {
+    "beaconId":"org.example.beacon.v2",
+    "apiVersion": "v2.0",
+    "returnedSchemas": [
+      { "entityType": "EntryTypeB", "schema": "entry-typeB-schema-v1.0"}
```
Comment (Tom-Shorter) [inline responses/examples-fullDocuments/beaconCollectionsResponse-MID-example.json:7]: This should probably match the `requestedSchemas` value to make this example clearer.

**Review by Tom-Shorter — CHANGES_REQUESTED** [review]
If the user only requests the dataset schema then should the implementer not display cohort responses and vice versa?

I think it make more sense to define a collections schema which is a list of cohort + dataset schemas and this is the requested/returned schema type

**Comment by mbaudis** [comment]
+1 AFAIK; i.e. a necessary abstraction wrapper for different types of "collections" (datasets, cohorts, studies ...). I'm not digging into the implementation right now (though it looks goodish to me).

File: responses/examples-fullDocuments/beaconCollectionsResponse-MID-example.json (hunk 1 above)
Comment (jrambla) [reply responses/examples-fullDocuments/beaconCollectionsResponse-MID-example.json]: I'll doublecheck

**Comment by jrambla** [comment]
@Tom-Shorter I was thinking in a filter by entry type if you are only interested in one or some.
As we have now the ontology term that "defines" the entry type, we can do that.
We can also filter by collection content type (e.g. "return only datasets containing biosamples").
I can create a examples of them.
Makes sense?

**Comment by jrambla** [comment]
By the way, as the example happens in the Framework, it doesn't "have" Cohorts.
Actually, it has no entry types defined, only in the examples.
Therefore, the only "definition" we have for a "generic collection" type, so far, is the entry type definition...

**Comment by mbaudis** [comment]
+1 @jrambla We actually use "collations", which cover things like cohorts, aggregated data from or used for individual studise etc. I think this can get co-opted (we make a difference regarding datasets which we still use as physical separators - but actually all the relevant data is in one).

That being said: Do we keep `/datasets` as a separate endpoint?

**Comment by Tom-Shorter** [comment]
@jrambla , using filters does make sense to define which collection types should be returned and keeping the requested/returned schemas as they currently are allows users to choose between any schemas defined for a collection whereas adding the collections schema would take this option away.

Maybe in the beacon configuration in the models repo a section is added which details the type of collections available within the beacon (datasets/cohorts, etc.) as well as what is currently there which describes the data contained within the collections.

**Comment by jrambla** [comment]
@Tom-Shorter what you are commenting is not what we already have in Entry types with the attribute `aCollectionOf`?
Do you miss something else?

**Comment by jrambla** [comment]
@mbaudis, yes, we have `/datasets` and `/cohorts` in the Beacon v2 Model. But not in the Framework. 
In Framework we have the option to define an Entry Type and to declare it as a collection of other entry types.
So you can end up defining  datasets, collections, cohorts, collations, etc.
In Beacon v1 Model and in the Template example, we use `/datasets`, as it seems natural, but not cohorts.

**Comment by Tom-Shorter** [comment]
Yes, I forgot about the `aCollectionOf` attribute which accomplishes basically what I had said. I prefer things to be more explicit though, such as collections being returned under a collections property and things like g-variants being returned under a dataType property like below (edited basic beaconConfiguration.json file):

`{
	"$schema": "https://raw.githubusercontent.com/ga4gh-beacon/beacon-framework-v2/main/configuration/beaconConfigurationSchema.json",

	"maturityAttributes": {
		"productionStatus": "DEV"
	},
	"securityAttributes": {
		"defaultGranularity": "record",
		"securityLevels": ["PUBLIC", "CONTROLLED"]
	},
	"entryTypes": {
                "collections": {
		        "dataset": {
			        "id": "dataset",
			        "name": "Dataset",
			        "ontologyTermForThisType": {
				        "id": "NCIT:C47824",
				        "label": "Data set"
			        },
			        "partOfSpecification": "Beacon v2.0.0-draft.3",
			        "description": "A Dataset is a collection of records, like rows in a database or cards in a cardholder.",
			        "defaultSchema": {
				        "id": "ga4gh-beacon-dataset-v2.0.0-draft.3",
				        "name": "Default schema for datasets",
				        "referenceToSchemaDefinition": "./datasets/defaultSchema.json",
				        "schemaVersion": "v2.0.0-draft.3"
			        },
			        "aCollectionOf": [{ "id": "genomicVariant", "name": "Genomic Variants" }],
			        "additionalSupportedSchemas": []
		        }
                },
                "dataTypes":{
                          "genomicVariant": {
			          "id": "genomicVariant",
			          "name": "Genomic Variants",
			          "ontologyTermForThisType": {
				          "id": "SO:0000735",
				          "label": "sequence_location"
			          },
			          "partOfSpecification": "Beacon v2.0.0-draft.3",
			          "description": "The location of a sequence.",
			          "defaultSchema": {
				          "id": "ga4gh-beacon-variant-v2.0.0-draft.3",
				          "name": "Default schema for a genomic variation",
				          "referenceToSchemaDefinition": "./genomicVariations/defaultSchema.json",
				          "schemaVersion": "v2.0.0-draft.3"
			          },
			          "additionallySupportedSchemas": []
		          }
                }
	}
}`

This is definitely personal preference though and I have no problems if the configuration file is left as is.

Overall I am happy with the state of this pull request, it does what it needs to do so I have approved it. We can come back to any of the other proposed ideas at a later date if needed.

**Review by Tom-Shorter — APPROVED** [review]

**Comment by jrambla** [comment]
I see your point, and I would concur.
The reason we haven't used that approach is to allow for referring to the entry type using  ["entryTypes"]["_whatever-entry-type_"] w/o the need to know in advance if it is a collection or not...
As in other decisions in the Beacon design, it is based on a subjective appreciation of which option is less annoying, or more intuitive... you put the adjective or criteria here.
@Tom-Shorter, @sdelatorrep, @MrRobb any opinion on that?

**Comment by Tom-Shorter** [comment]
I have just thought of a use case where my suggestion wouldn't work, a beacon has data which are separated into X datasets, these datasets are then split into Y different groups (I'd like to call these super collections). 

A user would then be able to search for results based on the properties of the data, the datasets or the dataset groups and this hierarchy wouldn't neatly fit into the format I proposed as datasets would be both a collection and a data type.

With beacons model system scenarios like this are likely to arise so I think that the current format is actually better as it allows for more flexibility within the entry type hierarchy.

**Pull request merged**